src/UnitTests/Gui/Windows/CombinedCodeViewInteractorTests.cs

```diff
@@ -0,0 +1,257 @@
+                {
+                    EmitSpanWrapper(span, sb);
+                    sb.Append(span.GetText());
+                    EmitSpanWrapper(span, sb);
+                }
+                sb.AppendLine();
+            }
+            return sb.ToString();
+        }
+
+        private void EmitSpanWrapper(TextSpan span, StringBuilder sb)
+        {
+            if (span.Style == "kw")
+                sb.Append("'");
+            if (span.Style == "cmt")
+                sb.Append("rem ");
+            if (span.Style == "link")
+                sb.Append("_");
+            if (span.Style == "type")
+                sb.Append("`");
+        }
+
+        private void When_CombinedCodeViewCreated()
+        {
+            combinedCodeView = (CombinedCodeView) combinedCodeViewInteractor.CreateControl();
+        }
+
+        private void When_MovedTo(uint addr)
+        {
+            combinedCodeViewInteractor.SelectedAddress = Address32.Ptr32(addr);
+        }
+
+        private void Given_ImageSegment(uint addr, params byte[] bytes)
+        {
+            var mem = new MemoryArea(Address32.Ptr32(addr), bytes);
+            var seg = new ImageSegment(".text", mem, AccessMode.ReadWrite);
+            program.ImageMap.AddSegment(seg);
+        }
+
+        private void Given_StubProcedure(uint addr, uint size)
+        {
+            var address = Address32.Ptr32(addr);
+            var m = new ProcedureBuilder("fnTest");
+            m.Return();
+            this.proc = m.Procedure;
+            this.program.Procedures[address] = proc;
+
+            var item = new ImageMapBlock
+            {
+                Address = address,
+                Size = size,
+                Block = new Block(proc, "fakeBlock")
+            };
+            program.ImageMap.AddItemWithSize(address, item);
+        }
+
+        private void Given_ImageMapItem(uint addr, DataType dataType, string name)
+        {
+            var address = Address32.Ptr32(addr);
+            var item = new ImageMapItem
+            {
+                Address = address,
+                DataType = dataType,
+                Name = name,
+                Size = (uint)dataType.Size,
+            };
+            program.ImageMap.AddItemWithSize(address, item);
+        }
+
+        [Test(Description = "When a user browses to the procedure, we should see it")]
+        public void Ccvi_SetProcedure()
+        {
+            Given_ImageSegment(0x10,
+                0xC3, /* ret */
+                0x01, 0x02, 0x03, 0x04,
+                0xE8, 0x03, 0x00, 0x00, /* 1000 */
+                0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B,
+                0x0C, 0x0D, 0x0E, 0x0F);
+            Given_StubProcedure(0x10, 1);
+            Given_ImageMapItem(0x15, PrimitiveType.Int32, "iVar");
+            mr.ReplayAll();
+
+            When_CombinedCodeViewCreated();
+            combinedCodeViewInteractor.DisplayProcedure(this.program, this.proc);
```

Comment:
Since there is only one interactor involved let's just call the variable  `interactor` for simplicity